core/src/main/scala/JwtClaim.scala

```diff
@@ -19,39 +25,105 @@ case class JwtClaim(
-  def issuedNow: JwtClaim = this.copy(issuedAt = Option(JwtTime.nowSeconds))
+  def about(subject: String): JwtClaim = {
+    JwtClaim(content, issuer, Option(subject), audience, expiration, notBefore, issuedAt, jwtId)
+  }
+
+  def withId(id: String): JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, expiration, notBefore, issuedAt, Option(id))
+  }
+
+  def expiresIn(seconds: Long): JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, Option(JwtTime.nowSeconds + seconds), notBefore, issuedAt, jwtId)
+  }
+
+  def expiresAt(seconds: Long): JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, Option(seconds), notBefore, issuedAt, jwtId)
+  }
+
+  def expiresNow: JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, Option(JwtTime.nowSeconds), notBefore, issuedAt, jwtId)
+  }
+
+  def startsIn(seconds: Long): JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, expiration, Option(JwtTime.nowSeconds + seconds), issuedAt, jwtId)
+  }
+
+  def startsAt(seconds: Long): JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, expiration, Option(seconds), issuedAt, jwtId)
+  }
+
+  def startsNow: JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, expiration, Option(JwtTime.nowSeconds), issuedAt, jwtId)
+  }
+
+  def issuedIn(seconds: Long): JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, expiration, notBefore, Option(JwtTime.nowSeconds + seconds), jwtId)
+  }
+
+  def issuedAt(seconds: Long): JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, expiration, notBefore, Option(seconds), jwtId)
+  }
+
+  def issuedNow: JwtClaim = {
+    JwtClaim(content, issuer, subject, audience, expiration, notBefore, Option(JwtTime.nowSeconds), jwtId)
+  }
 
-  def isValid: Boolean = JwtTime.nowIsBetweenSeconds(this.notBefore, this.expiration)
-  def isValid(issuer: String): Boolean = this.issuer.exists(_ == issuer) && this.isValid
   def isValid(issuer: String, audience: String): Boolean = this.audience.exists(_ contains audience) && this.isValid(issuer)
+
+  def isValid(issuer: String): Boolean = this.issuer.contains(issuer) && this.isValid
+
+  def isValid: Boolean = JwtTime.nowIsBetweenSeconds(this.notBefore, this.expiration)
+
+  // equality code
+  def canEqual(other: Any): Boolean = other.isInstanceOf[JwtClaim]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: JwtClaim =>
+      (that canEqual this) &&
+        content == that.content &&
+        issuer == that.issuer &&
+        subject == that.subject &&
+        audience == that.audience &&
+        expiration == that.expiration &&
+        notBefore == that.notBefore &&
+        issuedAt == that.issuedAt &&
+        jwtId == that.jwtId
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(content, issuer, subject, audience, expiration, notBefore, issuedAt, jwtId)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
```

Comment:
Is that the normal code for a `case class`? No idea where that `31` comes from.

Reply:
See comment above